pkg/controller.v1/tensorflow/tfjob_controller.go

```diff
@@ -21,55 +21,127 @@ import (
+	tfJobCreatedReason = "TFJobCreated"
+	// tfJobSucceededReason is added in a tfjob when it is succeeded.
+	tfJobSucceededReason = "TFJobSucceeded"
+	// tfJobRunningReason is added in a tfjob when it is running.
+	tfJobRunningReason = "TFJobRunning"
+	// tfJobFailedReason is added in a tfjob when it is failed.
+	tfJobFailedReason = "TFJobFailed"
+	// tfJobRestarting is added in a tfjob when it is restarting.
+	tfJobRestartingReason = "TFJobRestarting"
+
+	failedMarshalTFJobReason  = "InvalidTFJobSpec"
+	FailedDeleteJobReason     = "FailedDeleteJob"
+	SuccessfulDeleteJobReason = "SuccessfulDeleteJob"
+
+	controllerName = "tf-operator"
+
+	// labels for pods and servers.
+	tfReplicaTypeLabel  = "replica-type"
+	tfReplicaIndexLabel = "replica-index"
+	labelGroupName      = "group-name"
+	// Deprecated label for backwards compatibility. Has to be removed
+	labelTFJobName = "tf-job-name"
+	// volcanoTaskSpecKey task spec key used in pod annotation when EnableGangScheduling is true
+	volcanoTaskSpecKey = "volcano.sh/task-spec"
+
+	// gang scheduler name.
+	gangSchedulerName = "volcano"
+	// tfConfig is the environment variable name of TensorFlow cluster spec.
+	tfConfig = "TF_CONFIG"
+	// exitedWithCodeReason is the normal reason when the pod is exited because of the exit code.
+	exitedWithCodeReason = "ExitedWithCode"
+	// podTemplateRestartPolicyReason is the warning reason when the restart
+	// policy is set in pod template.
+	podTemplateRestartPolicyReason = "SettedPodTemplateRestartPolicy"
+	// podTemplateSchedulerNameReason is the warning reason when other scheduler name is set
+	// in pod templates with gang-scheduling enabled
+	podTemplateSchedulerNameReason = "SettedPodTemplateSchedulerName"
+	// gangSchedulingPodGroupAnnotation is the annotation key used by batch schedulers
+	gangSchedulingPodGroupAnnotation = "scheduling.k8s.io/group-name"
 )
 
 var (
-	defaultCleanPodPolicy = commonv1.CleanPodPolicyNone
+	tfJobsSuccessCount = promauto.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "tf_operator_jobs_successful_total",
+			Help: "Counts number of TF jobs successful",
+		},
+		[]string{"job_namespace"},
+	)
+	tfJobsFailureCount = promauto.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "tf_operator_jobs_failed_total",
+			Help: "Counts number of TF jobs failed",
+		},
+		[]string{"job_namespace"},
+	)
+	tfJobsCreatedCount = promauto.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "tf_operator_jobs_created_total",
+			Help: "Counts number of TF jobs created",
+		},
+		[]string{"job_namespace"},
+	)
+	tfJobsDeletedCount = promauto.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "tf_operator_jobs_deleted_total",
+			Help: "Counts number of TF jobs deleted",
+		},
+		[]string{"job_namespace"},
+	)
+	tfJobsRestartCount = promauto.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "tf_operator_jobs_restarted_total",
+			Help: "Counts number of TF jobs restarted",
+		},
+		[]string{"job_namespace"},
+	)
```

Comment:
Do we need this since we have [`common/metrics`](https://github.com/kubeflow/tf-operator/blob/master/pkg/common/metrics.go) ? cc @deepak-muley

Reply:
This needs to be removed

Reply:
correct these are stale counters. we only use the counters mentioned in common/metrics.go

Reply:
Good catch. Might've been due to bad rebasing on my end.